vortex-dtype/src/arrow.rs

```diff
@@ -150,73 +168,148 @@ impl DType {
         for (field_name, field_dtype) in struct_dtype.names().iter().zip(struct_dtype.fields()) {
             builder.push(FieldRef::from(Field::new(
                 field_name.to_string(),
-                field_dtype.to_arrow_dtype()?,
+                field_dtype.to_arrow_field()?.data_type().clone(),
                 field_dtype.is_nullable(),
             )));
         }
 
         Ok(builder.finish())
     }
 
-    /// Returns the Arrow [`DataType`] that best corresponds to this Vortex [`DType`].
-    pub fn to_arrow_dtype(&self) -> VortexResult<DataType> {
```

Comment:
I think you'll want to keep both of these actually. I had a PR to converted to DataType via Field, and creating and unwrapping the field was non-trivial due to the number of Arc's involved.